src/data/cel/strings.rs

```diff
@@ -0,0 +1,374 @@
+use cel_interpreter::extractors::{Arguments, This};
+use cel_interpreter::{ExecutionError, ResolveResult, Value};
+use std::sync::Arc;
+
+pub fn char_at(This(this): This<Arc<String>>, arg: i64) -> ResolveResult {
+    match this.chars().nth(arg as usize) {
+        None => Err(ExecutionError::FunctionError {
+            function: "String.charAt".to_owned(),
+            message: format!("No index {arg} on `{this}`"),
+        }),
+        Some(c) => Ok(c.to_string().into()),
+    }
+}
+
+pub fn index_of(
+    This(this): This<Arc<String>>,
+    arg: Arc<String>,
+    Arguments(args): Arguments,
+) -> ResolveResult {
+    match args.len() {
+        1 => match this.find(&*arg) {
+            None => Ok((-1).into()),
+            Some(idx) => Ok((idx as u64).into()),
+        },
+        2 => {
+            let base = match args[1] {
+                Value::Int(i) => i as usize,
+                Value::UInt(u) => u as usize,
+                _ => {
+                    return Err(ExecutionError::FunctionError {
+                        function: "String.indexOf".to_owned(),
+                        message: format!(
+                            "Expects 2nd argument to be an Integer, got `{:?}`",
+                            args[1]
+                        ),
+                    })
+                }
+            };
+            if base >= this.len() {
+                return Ok((-1).into());
+            }
+            match this[base..].find(&*arg) {
+                None => Ok((-1).into()),
+                Some(idx) => Ok(Value::UInt((base + idx) as u64)),
+            }
+        }
+        _ => Err(ExecutionError::FunctionError {
+            function: "String.indexOf".to_owned(),
+            message: format!("Expects 2 arguments at most, got `{args:?}`!"),
+        }),
+    }
+}
+
+pub fn last_index_of(
+    This(this): This<Arc<String>>,
+    arg: Arc<String>,
+    Arguments(args): Arguments,
+) -> ResolveResult {
+    match args.len() {
+        1 => match this.rfind(&*arg) {
+            None => Ok((-1).into()),
+            Some(idx) => Ok((idx as u64).into()),
+        },
+        2 => {
+            let base = match args[1] {
+                Value::Int(i) => i as usize,
+                Value::UInt(u) => u as usize,
+                _ => {
+                    return Err(ExecutionError::FunctionError {
+                        function: "String.lastIndexOf".to_owned(),
+                        message: format!(
+                            "Expects 2nd argument to be an Integer, got `{:?}`",
+                            args[1]
+                        ),
+                    })
+                }
+            };
+            if base >= this.len() {
+                return Ok((-1).into());
+            }
+            match this[base..].rfind(&*arg) {
+                None => Ok((-1).into()),
+                Some(idx) => Ok(Value::UInt(idx as u64)),
```

Comment:
Shouldn't this be `base +` similar to `index_of`?
```suggestion
                Some(idx) => Ok(Value::UInt(base + idx as u64)),
```

Reply:
It doesn't try... and tests won't pass (the tests are from the doc et al from the golang version)
That being said, when I read the expectations for that method, I failed to understand them

Reply:
Sure, thanks for [this link](https://github.com/google/cel-go/blob/master/ext/strings.go), I see this is expected